src/apps/something-similar/SomethingSimilarList.tsx

```diff
@@ -25,15 +26,23 @@ const SomethingSimilarList: FC<SomethingSimilarListProps & MaterialProps> = ({
     somethingSimilar: somethingSimilarLimit
   } = getRecommenderMaterialLimits();
   const [somethingSimilar, setSomethingSimilar] = useState<Work[] | null>(null);
-
   const [authorMaterials, setAuthorMaterials] = useState<Work[] | null>(null);
 
-  const [recommendView, setRecommendView] = useState<boolean>(true);
+  const [recommendView, setRecommendView] =
+    useState<RecommenderView>("similar");
   const { data: somethingSimilarData } = useRecommendFromFaustQuery({
     faust: id,
     limit: somethingSimilarLimit as number
   });
 
+  const setViewHandler = useCallback(
+    (inputView: RecommenderView) => {
+      window.history.pushState("", "", `?recommender-view=${inputView}`);
```

Comment:
Are you sure yow want to use pushState and thereby add to the browser history. Or do you want to rewrite the history? In that case you should use `setQueryParametersInUrl`.
But if you want to add to the history I think you should do following:

* Create a function for it in the [url helper file ](https://github.com/reload/dpl-react/blob/604f25f4a85f9ed6b9e6e9e542291aa7a9f76f18/src/core/utils/helpers/url.ts).
* I think it is a bad practise to hardcode query params like that. What if another url param has already been set by some other functionality? It is move safe to use the [searchParam API](https://github.com/reload/dpl-react/blob/604f25f4a85f9ed6b9e6e9e542291aa7a9f76f18/src/core/utils/helpers/url.ts#L31)

Reply:
I'll try setQueryParametersInUrl :)